### ISSUE TITLE
Spring ResponseEntity<T> equivalent

### ISSUE THREAD

**Issue by Fubuchi** [quarkiverse/quarkus-openapi-generator#825]
I need the generated java code include both response body and headers.

I know there is `return-response` option to return `javax.ws.rs.core.Response` but that class is not generic and requires casting all the time.

Is there anything similar to spring `ResponseEntity<T>` in quarkus?

**Comment by ricardozanini** [comment]
Something like this? https://stackoverflow.com/questions/74646006/theres-a-alternative-for-responseentity-in-quarkus

**Comment by Fubuchi** [comment]
Yes. Is there any way to generate `RestResponse<T>` instead of `javax.ws.rs.core.Response` ?

**Comment by ricardozanini** [comment]
No, but you can send a PR proposal extending this use case:
https://docs.quarkiverse.io/quarkus-openapi-generator/dev/client.html#returning-response-objects

So, one can have:

```properties
quarkus.openapi-generator.codegen.spec.my_openapi_yaml.return-response=RestResponse
```

or 

```properties
quarkus.openapi-generator.codegen.spec.my_openapi_yaml.return-response=Response
```

But still fallback to 

```properties
quarkus.openapi-generator.codegen.spec.my_openapi_yaml.return-response=true
```

To not break anything.
